### PR TITLE
Implemented the JAX BLSTM example using Pytorch

### PR DESCRIPTION
**Pull request by 1liale**
Model size: (batch_size = 256, seq_len = 128, embed_size = input_size = hidden_size = 192)

Steps Followed:
```
1. Load in the SNLI datasets

2. Tokenize it with roberta-base tokenizer with seq_len = 128

For 1-2 follow the `mlax/examples/lstm.ipynb`

3. Put them in PyTorch dataloaders.

For 3 follow mlax/examples/resnet/resnet_reference.ipynb

4. Create an LSTM model with parameters: vocab_size, embed_size, dropout
- Word embedding layer - torch.nn.Embed of vocab_size and embed_size
- Type embedding layer - torch.nn.Embed with a vocab_size of 2 and embed_size of embed_size, one embedding for question, one for answer

- Element-wise add word embedding and type embedding to get actual embedding
- Bidirectional LSTM with dropout
- Linear projection to make output shape equal to input shape (read torch docs, bidirectional output embedding size is doubled, why?)
- Another bidirectional LSTM with dropout and linear projection
- Another bidirecitional LSTM with dropout and linear projection
- Flatten (reshape) output from (batch_size, seq_len, embed_size) to (batch_size, *)
- Linear projection to 3 outputs

5. Define loss function (cross-entropy) and optimizer (adamw)

6. Define training and testing loops

7. Train model with embed_size = 192 and dropout = 0.1.
```

* Issue To reolve: masking outputs after the output_layer

### CODE REVIEW

**Comment by zongyf02** [comment]
Retrained after a hyperparameter sweep. Merged in via #24.
